src/pages/tutorials/smart-rollups/optimize.md

```diff
@@ -0,0 +1,68 @@
+---
+id: optimize
+title: "Part 4: Optimizing the kernel"
+lastUpdated: 11th October 2023
+---
+
+The kernel must be efficient with space and processing power because there is a size limit for kernels.
+In these steps, you optimize the kernel:
+
+1. Run this command to print the current size of the kernel:
+
+   ```bash
+   du -h target/wasm32-unknown-unknown/debug/hello_world_kernel.wasm
+   ```
+
+   You can run this command inside or outside of the Docker container.
+
+   The size of the compiled kernel and its dependencies may be 18MB or more, which is too large to deploy.
+
+1. In a terminal window outside of the Docker container, run the `wasm-strip` command to reduce the size of the kernel:
+
+   ```bash
+   wasm-strip target/wasm32-unknown-unknown/debug/hello_world_kernel.wasm
+   ```
+
+   This command removes WebAssembly code that is not necessary to run rollups.
+   You must run this command outside of the Docker container because it does not have the `wasm-strip` command.
+
+1. Run the `du` command again to see the new size of the kernel:
+
+   ```bash
+   du -h target/wasm32-unknown-unknown/debug/hello_world_kernel.wasm
+   ```
+
+   The size of the kernel is smaller now.
+   Note that the changes that you make to the kernel outside of the Docker container also appear in the container and vice versa because the folder is mounted with the Docker `--volume` argument.
+
+   To optimize the kernel further, you can convert it to an _installer kernel_, which includes only enough information to start the kernel, like an installation program.
+   This installer kernel keeps the rest of its logic and data in separate files called _preimages_.
```

Comment:
Not really - the preimages are _literally_ the original kernel. The install kernel effectively 'downloads them' (reveals the preimages) and writes them to `/kernel/boot.wasm` and then exits - the new kernel automatically starts.

Reply:
Updated to say:
> To optimize the kernel further, you can convert it to an _installer kernel_, which includes only enough information to start the kernel, like an installation program. To do this, you store the kernel logic and data in separate files called _preimages_. The installer kernel downloads them and uses them to restore the original kernel.

Reply:
Maybe very slightly change to:

To get the kernel running, you can use the installer kernel, which includes only enough information to install your original kernel. To do this, your kernel is split up and stored in separate files called preimages. The installer kernel then requests them and reconstructs the original kernel, before upgrading to it.

Reply:
> To get the kernel running with an even smaller size, you can use the installer kernel, which includes only enough information to install your original kernel. To do this, your kernel is split up and stored in separate files called preimages. Then you run the installer kernel, it requests these files and reconstructs the original kernel.